python/ray/data/_internal/planner/map_rows.py

```diff
@@ -21,7 +21,15 @@ def fn(
         for block in blocks:
             block = BlockAccessor.for_block(block)
             for row in block.iter_rows():
-                output_buffer.add(row_fn(row))
+                item = row_fn(row)
+                if context.strict_mode and not isinstance(item, dict):
```

Comment:
The `isinstance(item, dict)` test is unfortunately too strict and can lead to frustration. Here is and example for when this happens:

```
from datasets import load_dataset
dataset = load_dataset("rotten_tomatoes", split="train")
ds = ray.data.from_huggingface(dataset)

from transformers import AutoTokenizer
from datasets import load_dataset

tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")

def tokenization(example):
    return tokenizer(example["text"])

ds.map(tokenization).show()
```

The error is:
```
2023-04-16 23:35:52,110	INFO streaming_executor.py:87 -- Executing DAG InputDataBuffer[Input] -> TaskPoolMapOperator[Map]
2023-04-16 23:35:52,110	INFO streaming_executor.py:88 -- Execution config: ExecutionOptions(resource_limits=ExecutionResources(cpu=None, gpu=None, object_store_memory=None), locality_with_output=False, preserve_order=False, actor_locality_enabled=True, verbose_progress=False)
(Map pid=92025) 2023-04-16 23:35:52,315 INFO worker.py:844 -- Task failed with retryable exception: TaskID(d6dc98840c3203cdffffffffffffffffffffffff01000000).                          
(Map pid=92025) Traceback (most recent call last):                                                                                                                                     
(Map pid=92025)   File "python/ray/_raylet.pyx", line 659, in ray._raylet.execute_dynamic_generator_and_store_task_outputs                                                             
(Map pid=92025)   File "python/ray/_raylet.pyx", line 2746, in ray._raylet.CoreWorker.store_task_outputs                                                                               
(Map pid=92025)   File "/Users/pcmoritz/ray/python/ray/data/_internal/execution/operators/map_operator.py", line 389, in _map_task                                                     
(Map pid=92025)     for b_out in fn(iter(blocks), ctx):                                                                                                                                
(Map pid=92025)   File "/Users/pcmoritz/ray/python/ray/data/_internal/execution/legacy_compat.py", line 292, in do_map                                                                 
(Map pid=92025)     yield from block_fn(blocks, ctx, *fn_args, **fn_kwargs)                                                                                                            
(Map pid=92025)   File "/Users/pcmoritz/ray/python/ray/data/_internal/planner/map_rows.py", line 26, in fn                                                                             
(Map pid=92025)     raise ValueError(                                                                                                                                                  
(Map pid=92025) ValueError: Error validating {'input_ids': [101, 1996, 2600, 2003, 16036, 2000, 2022, 1996, 7398, 2301, 1005, 1055, 2047, 1000, 16608, 1000, 1998, 2008, 2002, 1005, 1055, 2183, 2000, 2191, 1037, 17624, 2130, 3618, 2084, 7779, 29058, 8625, 13327, 1010, 3744, 1011, 18856, 19513, 3158, 5477, 4168, 2030, 7112, 16562, 2140, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}: Standalone Python objects are not allowed in strict mode. To return Python objects from map(), wrap them in a dict, e.g., return `{'item': item}` instead of just `item`.
```

Reply:
The type here is:
```
>>> type(tokenizer(ds.take(1)[0]["text"]))
<class 'transformers.tokenization_utils_base.BatchEncoding'>
```
which inherits from `UserDict`.

Reply:
Even after I fix this, there is another error, can you have a look?
```
2023-04-17 00:05:15,482	WARNING dataset.py:4610 -- The `map`, `flat_map`, and `filter` operations are unvectorized and can be very slow. Consider using `.map_batches()` instead.
2023-04-17 00:05:15,483	INFO dataset.py:2293 -- Tip: Use `take_batch()` instead of `take() / show()` to return records in pandas or numpy batch format.
2023-04-17 00:05:15,485	INFO streaming_executor.py:87 -- Executing DAG InputDataBuffer[Input] -> TaskPoolMapOperator[Map]
2023-04-17 00:05:15,486	INFO streaming_executor.py:88 -- Execution config: ExecutionOptions(resource_limits=ExecutionResources(cpu=None, gpu=None, object_store_memory=None), locality_with_output=False, preserve_order=False, actor_locality_enabled=True, verbose_progress=False)
(Map pid=95133) 2023-04-17 00:05:17,445 INFO worker.py:844 -- Task failed with retryable exception: TaskID(e386a0ca68f0662bffffffffffffffffffffffff01000000).                          
(Map pid=95133) Traceback (most recent call last):                                                                                                                                     
(Map pid=95133)   File "python/ray/_raylet.pyx", line 659, in ray._raylet.execute_dynamic_generator_and_store_task_outputs                                                             
(Map pid=95133)   File "python/ray/_raylet.pyx", line 2746, in ray._raylet.CoreWorker.store_task_outputs                                                                               
(Map pid=95133)   File "/Users/pcmoritz/ray/python/ray/data/_internal/execution/operators/map_operator.py", line 389, in _map_task                                                     
(Map pid=95133)     for b_out in fn(iter(blocks), ctx):                                                                                                                                
(Map pid=95133)   File "/Users/pcmoritz/ray/python/ray/data/_internal/execution/legacy_compat.py", line 292, in do_map                                                                 
(Map pid=95133)     yield from block_fn(blocks, ctx, *fn_args, **fn_kwargs)                                                                                                            
(Map pid=95133)   File "/Users/pcmoritz/ray/python/ray/data/_internal/planner/map_rows.py", line 38, in fn                                                                             
(Map pid=95133)     yield output_buffer.next()                                                                                                                                         
(Map pid=95133)   File "/Users/pcmoritz/ray/python/ray/data/_internal/output_buffer.py", line 75, in next                                                                              
(Map pid=95133)     accessor = BlockAccessor.for_block(block)                                                                                                                          
(Map pid=95133)   File "/Users/pcmoritz/ray/python/ray/data/block.py", line 446, in for_block                                                                                          
(Map pid=95133)     raise ValueError(                                                                                                                                                  
(Map pid=95133) ValueError: Error validating [{'input_ids': [101, 1996, 2600, 2003, 16036, 2000, 2022, 1996, 7398, 2301, 1005, 1055, 2047, 1000, 16608, 1000, 1998, 2008, 2002, 1005, 1055, 2183, 2000, 2191, 1037, 17624, 2130, 3618, 2084, 7779, 29058, 8625, 13327, 1010, 3744, 1011, 18856, 19513, 3158, 5477, 4168, 2030, 7112, 16562, 2140, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 1996, 9882, 2135, 9603, 13633, 1997, 1000, 1996, 2935, 1997, 1996, 7635, 1000, 11544, 2003, 2061, 4121, 2008, 1037, 5930, 1997, 2616, 3685, 23613, 6235, 2522, 1011, 3213, 1013, 2472, 2848, 4027, 1005, 1055, 4423, 4432, 1997, 1046, 1012, 1054, 1012, 1054, 1012, 23602, 1005, 1055, 2690, 1011, 3011, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 4621, 2021, 2205, 1011, 8915, 23267, 16012, 24330, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 2065, 2017, 2823, 2066, 2000, 2175, 2000, 1996, 5691, 2000, 2031, 4569, 1010, 2001, 28518, 2003, 1037, 2204, 2173, 2000, 2707, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 19391, 2004, 2242, 4678, 1010, 2019, 3277, 3185, 2008, 1005, 1055, 2061, 7481, 1998, 10326, 2135, 5159, 2008, 2009, 2987, 1005, 1056, 2514, 2066, 2028, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 1996, 2143, 3640, 2070, 2307, 12369, 2046, 1996, 11265, 10976, 4588, 9273, 3388, 1997, 2035, 5888, 1011, 1011, 2130, 2216, 2040, 2031, 2584, 1996, 7619, 2327, 1997, 1996, 2208, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 4107, 2008, 4678, 5257, 1997, 4024, 1998, 2495, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 3383, 2053, 3861, 2412, 2081, 2038, 2062, 6719, 3662, 2008, 1996, 2346, 2000, 3109, 2003, 12308, 2007, 2204, 11174, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 20634, 2015, 4332, 1999, 1037, 10245, 7685, 9000, 2008, 14484, 2012, 1996, 7926, 1025, 2009, 1005, 1055, 2061, 12266, 2017, 2215, 2000, 5223, 2009, 1012, 2021, 2002, 5064, 8005, 2009, 2125, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 2202, 2729, 1997, 2026, 4937, 4107, 1037, 27150, 2135, 2367, 14704, 1997, 4004, 5988, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 2023, 2003, 1037, 2143, 2092, 4276, 3773, 1010, 3331, 1998, 4823, 4641, 1998, 2035, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 2054, 2428, 20096, 2055, 7968, 15239, 2015, 2003, 2049, 2659, 1011, 3145, 3737, 1998, 10218, 24605, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 1006, 19181, 4305, 3995, 2003, 1007, 2339, 2057, 2175, 2000, 1996, 5988, 1024, 2000, 2022, 7349, 2083, 1996, 3239, 1010, 1996, 2540, 1010, 1996, 2568, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 2028, 1997, 1996, 4602, 2155, 1011, 8048, 1010, 5913, 1011, 6172, 5691, 2412, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 4821, 1010, 2009, 29211, 2015, 1996, 4436, 2057, 2342, 3441, 2061, 2172, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 2019, 12580, 17075, 1005, 2040, 2626, 2009, 1005, 1999, 2029, 1996, 5891, 1997, 1996, 2087, 3297, 3166, 2040, 2412, 2973, 3310, 2046, 3160, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 27816, 2065, 15241, 2831, 2100, 4516, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 1037, 17743, 2176, 2086, 1999, 1996, 2437, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 1996, 3185, 1005, 1055, 22503, 1010, 4372, 2527, 13876, 12228, 5053, 2097, 8915, 27718, 2216, 5627, 2000, 15113, 2049, 16021, 26775, 23056, 15572, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 4107, 1037, 3052, 1997, 1996, 4840, 2250, 1997, 2995, 2061, 21850, 10074, 3370, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 1037, 16465, 1010, 26422, 1010, 29204, 2135, 2529, 6026, 2143, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 2007, 1037, 3459, 2008, 2950, 2070, 1997, 1996, 2327, 5889, 2551, 1999, 2981, 2143, 1010, 8403, 1004, 6429, 7336, 2149, 2138, 2009, 2003, 2061, 4297, 17417, 3726, 1010, 2061, 21657, 2135, 19142, 2055, 2129, 2057, 2175, 2055, 2256, 3268, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 1037, 14888, 1998, 17115, 2135, 23408, 24755, 6024, 3320, 1997, 13425, 1998, 28322, 2189, 3605, 2011, 5170, 3221, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 2025, 2005, 3071, 1010, 2021, 2005, 2216, 2007, 3183, 2009, 2097, 7532, 1010, 2009, 1005, 1055, 1037, 3835, 6712, 2013, 3115, 3185, 26966, 13258, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 7644, 1037, 2261, 2685, 2005, 2725, 2054, 2009, 2515, 2007, 1037, 4056, 1998, 2204, 1011, 18627, 2658, 2964, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 5681, 11463, 7716, 14672, 4588, 1010, 2009, 1005, 1055, 2036, 5186, 4621, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}, {'input_ids': [101, 6804, 2386, 5749, 102], 'token_type_ids': [0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1]}, ...
```